src/assisted_test_infra/test_infra/controllers/node_controllers/oci_controller.py

```diff
@@ -0,0 +1,149 @@
+        tfvars["master_disk_size_gib"] = self._config.master_disk / consts.GB
+        if self._config.worker_disk == consts.resources.DEFAULT_WORKER_DISK:
+            self._config.worker_disk = 50 * consts.GB
+
+        tfvars["worker_disk_size_gib"] = self._config.worker_disk / consts.GB
+        return tfvars
+
+    @property
+    def terraform_vm_name_key(self):
+        return "display_name"
+
+    @property
+    def terraform_vm_resource_type(self) -> str:
+        return "oci_core_instance"
+
+    def _get_provider_client(self) -> object:
+        oci_config = self._get_oci_config()
+
+        # Raise exception if failed
+        oci.config.validate_config(oci_config)
+        oci_client = oci.core.ComputeClient(oci_config)
+        oci_client.list_instances(self._config.oci_compartment_oicd)
+        self._virtual_network_client = oci.core.VirtualNetworkClient(self._get_oci_config())
+
+        return oci_client
+
+    def _get_provider_vm(self, tf_vm_name: str) -> Union[Instance, None]:
+        vm_attributes = self._get_vm(tf_vm_name)["attributes"]
+
+        oci_instances = self._provider_client.list_instances(self._config.oci_compartment_oicd).data
+        for instance in oci_instances:
+            if instance.id == vm_attributes["id"]:
+                return instance
+
+        raise ValueError(f"Can't find node with name: {tf_vm_name}")
+
+    def start_node(self, node_name: str, check_ips: bool) -> None:
+        """
+        :raises ValueError if node_name does not exist
+        """
+        instance = self._get_provider_vm(node_name)
+        if instance.lifecycle_state != OciInstanceState.RUNNING.value:
+            log.info(f"Powering on OCI instance {node_name}")
+            request = oci.core.models.UpdateInstanceDetails(action=OciInstanceAction.START)
+            self._provider_client.update_instance(instance.id, request)
+        else:
+            log.warning(
+                f"Attempted to power on node {node_name}, "
+                f"but the instance is already on - lifecycle_state={instance.lifecycle_state}"
+            )
+
+    def shutdown_node(self, node_name: str) -> None:
+        instance = self._get_provider_vm(node_name)
+        if instance.lifecycle_state != OciInstanceState.STOPPED.value:
+            log.info(f"Powering off OCI instance {node_name}")
+            request = oci.core.models.UpdateInstanceDetails(action=OciInstanceAction.STOP)
+            self._provider_client.update_instance(instance.id, request)
+        else:
+            log.warning(
+                f"Attempted to power off node {node_name}, "
+                f"but the instance is already off - lifecycle_state={instance.lifecycle_state}"
+            )
+
+    def restart_node(self, node_name: str) -> None:
+        instance = self._get_provider_vm(node_name)
+        log.info(f"Restarting OCI instance {node_name}")
+        request = oci.core.models.UpdateInstanceDetails(action=OciInstanceAction.SOFTRESET)
```

Comment:
```suggestion
        request = oci.core.models.UpdateInstanceDetails(action=OciInstanceAction.SOFTRESET)
```
Did it work for you? Each time I tried a softreset, it did not work properly. Maybe I was unlucky :thinking:

Reply:
Yes, but I retried it now and it seems that this syntax is not working.
I tested it with the following syntax and it worked:
```python
response = compute_client.instance_action(instance_id=instance.id, action="SOFTRESET")
```

Reply:
wierd :thinking: